ide/extexecution.base/src/org/netbeans/api/extexecution/base/ExplicitProcessParameters.java

```diff
@@ -0,0 +1,444 @@
+            if (priorityArguments == null) {
+                priorityArguments = new ArrayList<>();
+                /*
+                if (appendPriorityArgs == null) {
+                    appendPriorityArgs = true;
+                }
+                */
+            }
+        }
+        
+        /**
+         * Appends a single priority argument. {@code null} is ignored.
+         * @param a priority argument
+         * @return the builder
+         */
+        public Builder priorityArg(@NullAllowed String a) {
+            if (a == null) {
+                return this;
+            }
+            initPriorityArgs();
+            priorityArguments.add(a);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
+         * items in the list.
+         * @param args argument list
+         * @return the builder
+         */
+        public Builder priorityArgs(@NullAllowed List<String> args) {
+            if (args == null) {
+                return this;
+            }
+            initPriorityArgs();
+            args.forEach(this::priorityArg);
+            return this;
+        }
+
+        /**
+         * Appends arguments in the list. {@code null} is ignored as well as {@code null}
+         * items in the list.
+         * @param args argument list
+         * @return the builder
+         */
+        public Builder priorityArgs(@NullAllowed String... args) {
+            if (args == null) {
+                return this;
+            }
+            return priorityArgs(Arrays.asList(args));
+        }
+        
+        /**
+         * Changes the combining  mode for args. Setting to false instructs
+         * that all arguments that may precede should be discarded and the
+         * arguments provided by the built {@link ExplicitProcessParameters} are the only
+         * ones passed to the process.
+         * @param append true to append, false to replace
+         * @return the builder
+         */
+        public Builder appendArgs(boolean append) {
+            this.appendArgs = append;
+            return this;
+        }
+        
+        /**
+         * Changes the combining mode for priority args. Setting to false instructs
+         * that all arguments that may precede should be discarded and the
+         * priority arguments provided by the built {@link ExplicitProcessParameters} are the only
+         * ones passed to the process.
+         * @param append true to append, false to replace
+         * @return the builder
+         */
+        public Builder appendPriorityArgs(boolean append) {
+            this.appendPriorityArgs = append;
+            return this;
+        }
+
+        /**
+         * Defines a rank (position) for combining. The default rank is {@code 0}.
+         * @param rank rank of the instruction
+         * @return the builder
+         */
+        public Builder withRank(int rank) {
```

Comment:
Usually NetBeans API use "priority" and and not "rank", but I understand you have tough situation as "priority" is taken for "priority arguments" in this case.

Reply:
Addressed in 600178d5af